### PR TITLE
add new separator for role parsing logic

### PR DESCRIPTION
**Pull request by melionel**
# Description

add new separator for role parsing logic in chat string
- for system/user/assistant/function role, support single # prefix;
- for function's name/content role, support double # prefix

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-tools/promptflow/tools/common.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def parse_function_role_prompt(function_str):
 
 def parse_chat(chat_str):
     # openai chat api only supports below roles.
-    separator = r"(?i)\n+\s*(system|user|assistant|function)\s*:\s*\n"
+    separator = r"(?i)\n+\s*#?\s*(system|user|assistant|function)\s*:\s*\n"
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/common.py:101]: add some comments here to explain two things:
- `#` usage
- backward compatibility.

File: src/promptflow-tools/promptflow/tools/common.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def validate_functions(functions):
 
 
 def parse_function_role_prompt(function_str):
-    pattern = r"\n*name:\n\s*(\S+)\s*\n*content:\n(.*)"
+    pattern = r"\n*#{0,2}\s*name:\n\s*(\S+)\s*\n*#{0,2}\s*content:\n(.*)"
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/common.py:85]: also comment here, to denote 2 for heading 2 format and 0 for backward compatibility.

File: src/promptflow-tools/tests/test_common.py [hunk 3]
```diff
@@ -56,13 +56,23 @@ def test_parse_function_role_prompt(self):
...
+                {'role': 'system', 'content': 'this is my ##function:\ndef hello'}]),
             (" \n system:\nthis is my function:\ndef hello", [
                 {'role': 'system', 'content': 'this is my function:\ndef hello'}]),
+            (" \n # system:\nthis is my function:\ndef hello", [
+                {'role': 'system', 'content': 'this is my function:\ndef hello'}]),
             ("user:\nhi\nassistant:\nanswer\nfunction:\nname:\nn\ncontent:\nc", [
                 {'role': 'user', 'content': 'hi'},
                 {'role': 'assistant', 'content': 'answer'},
                 {'role': 'function', 'name': 'n', 'content': 'c'}]),
+            ("#user :\nhi\n #assistant:\nanswer\n# function:\nname:\nn\n##content:\nc", [
```
Comment (chjinche) [inline src/promptflow-tools/tests/test_common.py:69]: add a test case to for `##name`.

Comment (melionel) [reply src/promptflow-tools/tests/test_common.py:69]: update the existing test case to cover ##name

**Review by chjinche — APPROVED** [review]

**Review by DaweiCai — APPROVED** [review]

**Pull request merged**